### DISCUSSION TITLE
Guidelines for `must_use`?

### DISCUSSION THREAD

**Discussion by scottmcm** [rust-lang-nursery/api-guidelines#226] [API Guidelines]
I got assigned a pr (https://github.com/rust-lang/rust/pull/78202) to add this to something, and realized I didn't have a great sense of when something should or shouldn't be `must_use` -- in some ways most things could be.

Would be nice to have a checklist of things to consider.

This is especially true with `must_use` stable on functions now -- some people would like to put it on essentially everything, which seems like not what's desirable.  I suppose guidelines on this could also push feedback up to `lang` to change the rules for it...

**Comment by m-ou-se** [comment]
cc https://github.com/rust-lang/rust/issues/48926
